sdk/textanalytics/Azure.AI.TextAnalytics/tests/AnalyzeOperationTests.cs

```diff
@@ -0,0 +1,410 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.AI.TextAnalytics.Models;
+using NUnit.Framework;
+
+namespace Azure.AI.TextAnalytics.Tests
+{
+    public class AnalyzeOperationTests : TextAnalyticsClientLiveTestBase
+    {
+        public AnalyzeOperationTests(bool isAsync) : base(isAsync) { }
+
+        private static List<string> batchConvenienceDocuments = new List<string>
+        {
+            "Elon Musk is the CEO of SpaceX and Tesla.",
+            "Tesla stock is up by 400% this year."
+        };
+
+        private static List<TextDocumentInput> batchDocuments = new List<TextDocumentInput>
+        {
+            new TextDocumentInput("1", "Elon Musk is the CEO of SpaceX and Tesla.")
+            {
+                 Language = "en",
+            },
+            new TextDocumentInput("2", "Tesla stock is up by 400% this year.")
+            {
+                 Language = "en",
+            }
+        };
+
+        [Test]
+        public async Task AnalyzeOperationTest()
+        {
+            TextAnalyticsClient client = GetClient();
+
+            AnalyzeOperationOptions operationOptions = new AnalyzeOperationOptions()
+            {
+                KeyPhrasesTaskParameters = new KeyPhrasesTaskParameters()
+                {
+                    ModelVersion = "latest"
+                },
+                DisplayName = "AnalyzeOperationTest"
+            };
+
+            AnalyzeOperation operation = await client.StartAnalyzeOperationBatchAsync(batchConvenienceDocuments, "en", operationOptions);
+
+            await operation.WaitForCompletionAsync();
+
+            AnalyzeOperationResult resultCollection = operation.Value;
+
+            ExtractKeyPhrasesResultCollection result = resultCollection.KeyPhraseResult;
```

Comment:
Could you check that `RecognizeEntitiesResultCollection` and `RecognizePiiEntitiesResultCollection` are not null?

Reply:
They both will be null as they were not sent in options to have a value.

Reply:
Although that is true, in .NET we don't return/expose null collections. If no value was returned, then the collections are empty. This helps users write their code in a way where they don't get null reference exceptions.